src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'clear') {
+      currentState = {};
+    } else if (action.type === 'addProperties') {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === 'removeProperties') {
+      const newState = {};
+
+      for (const key in currentState) {
+        if (!action.keysToRemove.includes(key)) {
+          newState[key] = currentState[key];
+        }
+      }
+      currentState = newState;
+    }
```

Comment:
Instead of using a series of if-else statements, consider refactoring to use a switch statement for handling the different action types. This will improve readability and is more aligned with the checklist requirement for using a switch statement when you have a limited amount of conditions.

Reply:
Instead of using a series of if-else statements, consider refactoring to use a switch statement for handling the different action types.